modules/aidebug/cpp/aidebug.cpp

```diff
@@ -0,0 +1,129 @@
+﻿#include "../src/common/mmo.h"
+#include "../src/map/ai/ai_container.h"
+#include "../src/map/entities/mobentity.h"
+#include "../src/map/entities/npcentity.h"
+#include "../src/map/mob_modifier.h"
+#include "../src/map/utils/mobutils.h"
+#include "../src/map/utils/moduleutils.h"
+#include "../src/map/zone.h"
+#include "../src/map/zone_entities.h"
+
+#include <vector>
+
+class AIDebugModule : public CPPModule
+{
+    // std::vector<CBaseEntity*> m_entities;
+    std::map<uint16, std::vector<CBaseEntity*>> m_entities;
+    std::map<uint16, int>                       m_used;
+
+    void OnInit() override
+    {
+        TracyZoneScoped;
+        ShowInfo("AIDebug: Module Loaded");
+    }
+
+    void OnZoneTick(CZone* PZone) override
+    {
+        TracyZoneScoped;
+
+        ClearRenderedMobs(PZone);
+
+        auto mobList = PZone->GetZoneEntities()->GetMobList();
+        for (auto const& [targid, entity] : mobList)
+        {
+            auto mob = dynamic_cast<CMobEntity*>(entity);
+            if (entity->GetLocalVar("AIDEBUG_ENABLED") == 1)
+            {
+                if (m_entities.size() < 1)
+                    BuildDebugPool(PZone);
+                RenderPathForMob(PZone, mob);
+            }
+        }
+    }
+
+    void ClearRenderedMobs(CZone* PZone)
+    {
+        if (m_entities.find(PZone->GetID()) != m_entities.end() && m_entities[PZone->GetID()].size() < 1)
+            return;
+
+        for (auto i = m_used[PZone->GetID()]; i < m_entities[PZone->GetID()].size(); i++)
+        {
+            if (m_entities[PZone->GetID()][i]->status != STATUS_TYPE::DISAPPEAR)
+                m_entities[PZone->GetID()][i]->FadeOut();
+        }
+    }
+
+    void RenderPathForMob(CZone* PZone, CMobEntity* PMob)
+    {
+        std::vector<CBaseEntity*> validPlayers;
```

Comment:
You can typedef here, or you can define: `auto validPlayers = std::vector<CBaseEntity*>()`